src/features/collapsible-sidebar/CollapsibleSidebarNav.js

```diff
@@ -69,10 +70,6 @@ class CollapsibleSidebarNav extends React.Component<Props, State> {
         this.setScrollShadowState();
     }
 
-    componentDidUpdate() {
-        this.setScrollShadowState();
```

Comment:
`componentDidUpdate` won't execute If the content's height is changed.

There are alternative solutions to `onUpdate`.
- `MutationObserver`: We could observe `this.scrollRef.current.contentElement` and look for attributes changes. However, this is not extensible and requires this component to be aware of children's change. Also, even if we listen to `style` attributes, it cannot detect a height change after an animation (i.e. `transition`)
- `ontransitionend` event handler: `transitionend` event gets fired from the animation or scroll. Though it may be more performant compared to `onUpdate` since it fires less, it does not guarantee to handle height change.